actors/builtin/market/market_actor.go

```diff
@@ -217,7 +217,11 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 			deal.Proposal.Client = client
 
 			st.lockBalanceOrAbort(rt, client, deal.Proposal.ClientBalanceRequirement())
+			st.TotalLockedClientCollateral = big.Add(st.TotalLockedClientCollateral, deal.Proposal.ClientCollateral)
+			st.TotalLockedDealPayments = big.Add(st.TotalLockedDealPayments, deal.Proposal.TotalStorageFee())
```

Comment:
I think the state object should maintain the invariants, so this logic should be moved into the state methods. Probably that means changing the `lockBalanceOrAbort` to distinguish the two parts of client lock-up.

So how about `st.LockValue(party, amount, role)` where role is an enum of 3 options: provider collateral, client collateral, storage fee. Then we'll need three calls to that function here for all the lock-ups.

Note also that `lockBalanceOrAbort` is an example of the anti-pattern of passing the `rt` just for access to `Abortf`. Change it to return `(ok, error)` instead. The precondition check that the proposal's values are positive (`ErrIllegalArgument`) should move out to the caller, here. Then `ok=false` indicates `ErrInsufficientFunds`.

Reply:
@anorth I think I see hat you mean. I've introduced a `BalanceLockingReason` enum with values as 

* ClientCollateral
* ClientStorageFee
* ProviderCollateral

Both `st.LockFunds()` and `st.UnlockFunds` now take this enum so as to add/subtract to the corresponding state variables. 

Please note that I do not want to make the change of not aborting in "helper functions" as we plan to do it in #455 .